subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -82,43 +96,26 @@ int lll_hfclock_on(void)
 
 int lll_hfclock_on_wait(void)
 {
-	struct k_sem sem_clock_wait;
-	struct clock_control_async_data async_data = {
-		.cb = clock_ready,
-		.user_data = &sem_clock_wait,
-	};
-	int err;
-
-	k_sem_init(&sem_clock_wait, 0, 1);
-
-	err = clock_control_async_on(dev, CLOCK_CONTROL_NRF_SUBSYS_HF,
-				     &async_data);
-	LL_ASSERT(!err);
-
-	err = k_sem_take(&sem_clock_wait, K_MSEC(HFCLOCK_TIMEOUT_MS));
-
-	DEBUG_RADIO_XTAL(1);
-
-	return err;
+	return on_wait(&hf_state, HFCLOCK_TIMEOUT_MS);
```

Comment:
`lll_hfclock_on_wait` functionality is changed. This function needs to start hfclock in blocking mode. In the change it is only doing `k_sem_take` followed by `k_sem_give` in the `on_wait` function.

Reply:
right, misread it, i thought it's blocking waiting for already request clock. fixed.